cmd/genji/shell/command.go

```diff
@@ -366,3 +378,118 @@ func runDumpCmd(db *genji.DB, tables []string, w io.Writer) error {
 	_, err = fmt.Fprintln(w, "COMMIT;")
 	return err
 }
+
+// runSaveCommand saves the currently opened database in the given filename.
+// If a file already exists, it will be overwritten.
+func runSaveCmd(ctx context.Context, db *genji.DB, engineName string, path string) error {
+	tx, err := db.Begin(false)
+	if err != nil {
+		return err
+	}
+	defer tx.Rollback()
+
+	// Open the new database
+	var otherNg engine.Engine
+
+	switch engineName {
+	case "bolt":
+		otherNg, err = boltengine.NewEngine(path, 0660, nil)
+		if err != nil {
+			return err
+		}
+	case "badger":
+		otherNg, err = badgerengine.NewEngine(badger.DefaultOptions(path).WithLogger(nil))
+		if err != nil {
+			return err
+		}
+	default:
+		return fmt.Errorf("Can't use unknown engine %s", engineName)
+	}
+
+	otherDB, err := genji.New(ctx, otherNg)
```

Comment:
todo: use `otherDb.WithContext(ctx)` in order to enable cancelling long queries.